ocaml/xapi/xapi_pif.ml

```diff
@@ -737,10 +737,14 @@ let reconfigure_ipv6 ~__context ~self ~mode ~iPv6 ~gateway ~dNS =
   (* If we have an IPv6 address, check that it is valid and a prefix length is specified *)
   if iPv6 <> "" then
     Helpers.assert_is_valid_cidr `ipv6 "IPv6" iPv6 ;
-  if dNS <> "" then
-    List.iter
-      (fun address -> Helpers.assert_is_valid_ip `ipv6 "DNS" address)
-      (String.split ',' dNS) ;
+  (* Only set IPv6 dNS *)
+  let ipv6_dns_to_set =
+    List.filter
+      (fun dns_addr ->
+        Xapi_stdext_unix.Unixext.domain_of_addr dns_addr = Some Unix.PF_INET6
+      )
+      (String.split_on_char ',' dNS)
```

Comment:
Using `Astring.String.cuts ~sep:',' ~empty:false` would allow to match against empty lists, without worrying about lists with empty elements further down.

It would be better dNS would only be processed once instead of twice. An even further enhancement would be to use `let ipv4_dns, ipv6_set = List.partition (fun dns_addr -> ...)`, but this only works if there are 2 types of addresses here

Reply:
`dNS` is only processed once the 2nd time is the `PIF.DNS` field.